server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreStatsIT.java

```diff
@@ -111,29 +173,307 @@ public void testStatsResponseFromLocalNode() {
+            validateUploadStats(stats);
+        }
+        changeReplicaCount(1);
+        for (String node : nodes) {
+            RemoteStoreStatsRequestBuilder remoteStoreStatsRequestBuilder = client(node).admin()
+                .cluster()
+                .prepareRemoteStoreStats(INDEX_NAME, null);
+            remoteStoreStatsRequestBuilder.setLocal(true);
+            RemoteStoreStatsResponse response = remoteStoreStatsRequestBuilder.get();
+            assertTrue(response.getSuccessfulShards() > 0);
+            assertTrue(response.getRemoteStoreStats() != null && response.getRemoteStoreStats().length != 0);
+            for (RemoteStoreStats stat : response.getRemoteStoreStats()) {
+                ShardRouting routing = stat.getShardRouting();
+                RemoteRefreshSegmentTracker.Stats stats = stat.getStats();
+                if (routing.primary()) {
+                    validateUploadStats(stats);
+                } else {
+                    validateDownloadStats(stats);
+                }
+            }
         }
     }
 
-    private void indexDocs() {
-        // Indexing documents along with refreshes and flushes.
-        for (int i = 0; i < randomIntBetween(5, 10); i++) {
-            if (randomBoolean()) {
-                flush(INDEX_NAME);
-            } else {
-                refresh(INDEX_NAME);
+    public void testDownloadStatsCorrectnessSinglePrimarySingleReplica() throws Exception {
+        // Scenario:
+        // - Create index with single primary and single replica shard
+        // - Disable Refresh Interval for the index
+        // - Index documents
+        // - Trigger refresh and flush
+        // - Assert that download stats == upload stats
+        // - Repeat this step for random times (between 2 and 5)
+
+        // Create index with 1 pri and 1 replica
+        createIndex(INDEX_NAME, remoteStoreIndexSettings(1, 1));
+        // Disabling refresh interval
+        disableRefreshInterval();
+
+        // Index documents
+        for (int i = 1; i <= randomIntBetween(2, 5); i++) {
+            indexDocs(false);
+            // Running Flush & Refresh manually
+            flushAndRefresh(INDEX_NAME);
+
+            // Poll for RemoteStore Stats
+            assertBusy(() -> {
+                RemoteStoreStatsResponse response = client().admin().cluster().prepareRemoteStoreStats(INDEX_NAME, "0").get();
+                long uploadsStarted = 0, uploadsSucceeded = 0, uploadsFailed = 0;
+                long uploadBytesStarted = 0, uploadBytesSucceeded = 0, uploadBytesFailed = 0;
+                long downloadsStarted = 0, downloadsSucceeded = 0, downloadsFailed = 0;
+                long downloadBytesStarted = 0, downloadBytesSucceeded = 0, downloadBytesFailed = 0;
+                double uploadBytesAverage = 0, downloadBytesAverage = 0;
+                long lastUploadedSegmentSize = 0, lastDownloadedSegmentSize = 0;
```

Comment:
nit - can we assert directly from the response of the stats api?

Reply:
We are iterating through an array of stats objects here which has the stats of both primary as well as replica copies. This was needed to pick upload stats specifically from the primary copy's object and use it later on while comparing it with the replica copy stats. Not sure if this can be shortened further.